### PR TITLE
Remove scenario ticks and adjust export/import

### PR DESCRIPTION
**Pull request by ZehMatt**
Another NSF backport, this one might break replays.

### CODE REVIEW

File: src/openrct2/rct2/S6Importer.cpp [hunk 1]
```diff
@@ -480,6 +479,11 @@ class S6Importer final : public IParkImporter
 
         research_determine_first_of_type();
     }
+
+    void ConvertScenarioTicks()
```
Comment (ZehMatt) [inline src/openrct2/rct2/S6Importer.cpp:483]: Stray change

**Review by duncanspumpkin — APPROVED** [review]
All looks fine to me. Await @Gymnasiast  review prior to merge.

**Comment by Gymnasiast** [comment]
Would like to test a few things, e.g. stuff like https://github.com/OpenRCT2/OpenRCT2/issues/14734.

**Comment by ZehMatt** [comment]
> Would like to test a few things, e.g. stuff like #14734.

There are three things to watch out for with this PR
1. Guest time in park
2. Staff hire time
3. Litter creation time, this influences the park rating and is somewhat hard to test I guess.

But given the guest and staff times seemed fine at my end I guess litter will be also.

**Comment by Gymnasiast** [comment]
I don’t see code to properly import litter creation time in S6Importer, though. Is that intentional?

**Comment by ZehMatt** [comment]
> I don’t see code to properly import litter creation time in S6Importer, though. Is that intentional?

Nope it seems I forgot that one even tho I know I must do it, nice catch.

**Review by Gymnasiast — CHANGES_REQUESTED** [review]
Does not work correctly. The mechanics in this save all have a hire date of June, yr 528 instead of March, yr 1.

[Pakhuis - Derde Alter Ego - fixedMR.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/7190657/Pakhuis.-.Derde.Alter.Ego.-.fixedMR.sv6.zip)

**Comment by ZehMatt** [comment]
HireDate is used as an absolute time for  {MONTHYEAR} unlike guests which use a relative time, it should be good now.

File: src/openrct2/rct2/S6Importer.cpp [hunk 2]
```diff
@@ -1727,6 +1727,15 @@ template<> void S6Importer::ImportEntity<Vehicle>(const RCT12SpriteBase& baseSrc
     dst->IsCrashedVehicle = src->flags & RCT12_SPRITE_FLAGS_IS_CRASHED_VEHICLE_SPRITE;
 }
 
+static uint32_t AdjustScenarioToCurrentTicks(const rct_s6_data& s6, uint32_t tick)
+{
+    // Previously gScenarioTicks was used as a time point now its gCurrentTicks.
```
Comment (Gymnasiast) [inline src/openrct2/rct2/S6Importer.cpp:1732]:
```suggestion
    // Previously gScenarioTicks was used as a time point, now it’s gCurrentTicks.
```

File: src/openrct2/rct2/S6Importer.cpp [hunk 3]
```diff
@@ -1727,6 +1727,15 @@ template<> void S6Importer::ImportEntity<Vehicle>(const RCT12SpriteBase& baseSrc
...
+    // gCurrentTicks and gScenarioTicks are now exported as the same, older saves that have a different
+    // scenario tick must add account for the distance between the two.
```
Comment (Gymnasiast) [inline src/openrct2/rct2/S6Importer.cpp:1734]:
```suggestion
    // scenario tick must account for the difference between the two.
```

**Review by Gymnasiast — APPROVED** [review]
Still a bit apprehensive (if this breaks, it will be in subtle and hard-to-detect ways), but I haven’t found any more bugs.

**Pull request merged**